micro-rdk/src/common/credentials_storage.rs

```diff
@@ -174,42 +195,76 @@ impl RAMStorage {
 
 #[cfg(feature = "ota")]
 impl OtaMetadataStorage for RAMStorage {
-    type Error = Infallible;
+    type Error = RAMStorageError;
     fn has_ota_metadata(&self) -> bool {
         let inner_ref = self.0.lock().unwrap();
         inner_ref.ota_metadata.is_some()
     }
-    fn store_ota_metadata(&self, ota_metadata: OtaMetadata) -> Result<(), Self::Error> {
+    fn store_ota_metadata(&self, ota_metadata: &OtaMetadata) -> Result<(), Self::Error> {
         let mut inner_ref = self.0.lock().unwrap();
-        let _ = inner_ref.ota_metadata.insert(ota_metadata);
+        let _ = inner_ref.ota_metadata.insert(ota_metadata.clone());
         Ok(())
     }
     fn get_ota_metadata(&self) -> Result<OtaMetadata, Self::Error> {
         let inner_ref = self.0.lock().unwrap();
-        Ok(inner_ref.ota_metadata.clone().unwrap_or_default())
+        inner_ref
+            .ota_metadata
+            .clone()
+            .ok_or(RAMStorageError::NotFound)
     }
     fn reset_ota_metadata(&self) -> Result<(), Self::Error> {
         let _ = self.0.lock().unwrap().ota_metadata.take();
         Ok(())
     }
 }
+#[cfg(feature = "ota")]
+impl<Iterable, Storage: OtaMetadataStorage> OtaMetadataStorage for Iterable
+where
+    for<'a> &'a Iterable: IntoIterator<Item = &'a Storage>,
+    Storage::Error: From<EmptyStorageCollectionError>,
+{
+    type Error = Storage::Error;
+    fn has_ota_metadata(&self) -> bool {
+        self.into_iter().any(OtaMetadataStorage::has_ota_metadata)
+    }
+    fn get_ota_metadata(&self) -> Result<OtaMetadata, Self::Error> {
+        self.into_iter().fold(
```

Comment:
the idea here  is to use fold to call `get_ota_metadata` until Ok is returned. If all storage in a collection return an error only the last error is reported.
We don't have an iterator that behave like `try_fold` but actually short-circuit on the first Ok, so we will iterate through he whole collection even after a `get_ota_metadata` has succeeded.